File: sample/src/main/java/pl/droidsonroids/gif/sample/ImageSpanFragment.java
package pl.droidsonroids.gif.sample;

import android.graphics.drawable.Drawable;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.text.SpannableStringBuilder;
import android.text.Spanned;
import android.text.style.ImageSpan;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import pl.droidsonroids.gif.GifDrawable;

public class ImageSpanFragment extends BaseFragment implements Drawable.Callback {

	private EditText mEditText;

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		mEditText = new EditText(getActivity());
		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
			mEditText.setLayerType(View.LAYER_TYPE_SOFTWARE, null);
		}
		final GifDrawable gifDrawable = GifDrawable.createFromResource(getResources(), R.drawable.anim_flag_england);
		final SpannableStringBuilder ssb = new SpannableStringBuilder("test\ufffc");
		assert gifDrawable != null;
		gifDrawable.setBounds(0, 0, gifDrawable.getIntrinsicWidth(), gifDrawable.getIntrinsicHeight());
		gifDrawable.setCallback(this);
		ssb.setSpan(new ImageSpan(gifDrawable), ssb.length() - 1, ssb.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
		mEditText.getText().insert(0, ssb);
		return mEditText;
	}

	@Override
	public void invalidateDrawable(@NonNull Drawable who) {
		mEditText.invalidate();
	}

	@Override
	public void scheduleDrawable(@NonNull Drawable who, @NonNull Runnable what, long when) {
		mEditText.postDelayed(what, when);
	}

	@Override
	public void unscheduleDrawable(@NonNull Drawable who, @NonNull Runnable what) {
		mEditText.removeCallbacks(what);
	}
}

File: sample/src/main/java/pl/droidsonroids/gif/sample/ImageSpanFragment.kt
package pl.droidsonroids.gif.sample

import android.graphics.drawable.Drawable
import android.os.Bundle
import android.text.SpannableStringBuilder
import android.text.style.ImageSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import pl.droidsonroids.gif.GifDrawable

class ImageSpanFragment : Fragment(), Drawable.Callback {

    private var imageSpanTextView: TextView? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
        inflater.inflate(R.layout.image_span, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        imageSpanTextView = view.findViewById(R.id.imageSpanTextView)
        imageSpanTextView?.setLayerType(View.LAYER_TYPE_SOFTWARE, null)
        val gifDrawable = GifDrawable.createFromResource(resources, R.drawable.anim_flag_england)
        val stringBuilder = SpannableStringBuilder("test\ufffc")
        gifDrawable!!.setBounds(0, 0, gifDrawable.intrinsicWidth, gifDrawable.intrinsicHeight)
        gifDrawable.callback = this
        stringBuilder.setSpan(ImageSpan(gifDrawable), stringBuilder.length - 1, stringBuilder.length, 0)
        imageSpanTextView?.text = stringBuilder
    }

    override fun onDestroyView() {
        imageSpanTextView = null
        super.onDestroyView()
    }

    override fun invalidateDrawable(who: Drawable) {
        imageSpanTextView?.invalidate()
    }

    override fun scheduleDrawable(who: Drawable, what: Runnable, `when`: Long) {
        imageSpanTextView?.postDelayed(what, `when`)
    }

    override fun unscheduleDrawable(who: Drawable, what: Runnable) {
        imageSpanTextView?.removeCallbacks(what)
    }
}
